.github/workflows/tests.yml

```diff
@@ -0,0 +1,97 @@
+name: "Tests"
+
+on:
+  pull_request:
+    branches:
+      - "v*.*"
+      - "master"
+  push:
+    branches:
+      - "v*.*"
+      - "master"
+
+jobs:
+  phpunit:
+    name: "PHPUnit tests"
+    runs-on: "${{ matrix.os }}"
+
+    strategy:
+      fail-fast: true
+      matrix:
+        os:
+          - "ubuntu-20.04"
+        php-version:
+          - "7.4"
+          - "8.0"
+        mongodb-version:
+          - "4.4"
+        driver-version:
+          - "stable"
+        topology:
+          - "server"
+        include:
+          - os: "ubuntu-20.04"
+            php-version: "8.0"
+            mongodb-version: "4.4"
+            driver-version: "stable"
+            topology: "replica_set"
+          - os: "ubuntu-20.04"
+            php-version: "8.0"
+            mongodb-version: "4.4"
+            driver-version: "stable"
+            topology: "sharded_cluster"
+
+    steps:
+      - name: "Checkout"
+        uses: "actions/checkout@v2"
+        with:
+          fetch-depth: 2
+
+      - id: setup-mongodb
+        uses: mongodb-labs/drivers-evergreen-tools@master
```

Comment:
I assume versioning is a separate can of worms, since those tags would also apply to the entire repository and not just actions? OK with using `master` here, but just curious.

Reply:
Yes, this uses git tags, which apply to the whole repository. Since we don't version the repo at all I've decided to use master until we decide that we want to start tagging releases in that repository.